source/sonata_report.rst

```diff
@@ -86,6 +88,84 @@ For a full compartment reports, you need to specify both ``"sections": "all"`` a
         }
     }
 
+Extension of Compartment report
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable fine-grained control over which specific compartments are recorded, SONATA supports an extension using a ``compartment_sets.json`` file. This allows users to define sets of explicit compartment targets (e.g., ``dend[10](0.1)``) for selected node_id.
+
+File: ``compartment_sets.json``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This JSON file defines named sets of compartment targets. Each set specifies the population and a list of `(node_id, section_name, location)` tuples.
+
+.. code-block:: json
+
+  {
+    "example_compartment_set": {
+      "population": "S1nonbarrel_neurons",
+      "compartment_set": [
+        [0, "dend[10]", 0.1],
```

Comment:
Bringing everyone up to speed from the discussion in the issue and comments [here](https://github.com/openbraininstitute/sonata-extension/issues/4#issuecomment-2904447979):  
1. @mgeplf's suggestion is to have 
structure like this `[1, "dend", [[10, 0.2]]]` i.e. `[node_id, section_type, [[section_number, compartment_location]]]`
2. Another suggestion when recording from multiple compartments of the same section is to use `[0, "dend", [[3, 0.1], [5, 0.5], [9, 0.5]]],`

Reply:
@mgeplf Based on the above comments, should we keep the original structure `[0, "dend[10]", 0.1],`? 

And if it is fine with you, can you approve this PR?